src/rprblender/nodes/rpr_nodes.py

```diff
@@ -1513,3 +1513,95 @@ def export_hybrid(self):
                 return None
 
             return self.export()
+
+
+class RPRShaderNodeToon(RPRShaderNode):
+    ''' A toon shader using both the RPR Toon Shader and Ramp node '''
+    bl_label = 'RPR Toon'
+
+    def advanced_changed(self, context):
+        ramp_sockets = ['Shadow Color', "Mid Level","Mid Color", "Highlight Level", "Highlight Color"]
+        mix_sockets = ["Mid Level mix", "Highlight Level mix"]
+        if self.show_advanced:
+            self.inputs['Color'].enabled = Fals
+            for socket in ramp_sockets:
+                self.inputs[socket].enabled = True
+            for socket in mix_sockets:
+                self.inputs[socket].enabled = self.show_mix_levels
+        else:
+            for socket in ramp_sockets + mix_sockets:
+                self.inputs[socket].enabled = False
+            self.inputs['Color'].enabled = True
+
+    show_advanced: BoolProperty(name="Advanced", default=False, update=advanced_changed)
+    show_mix_levels: BoolProperty(name="Mix Levels", default=False, update=advanced_changed)
+
+    def init(self, context):
+        # Adding input sockets with default_value or hide_value properties.
+        # Here we use Blender's native node sockets
+        self.inputs.new('NodeSocketColor', "Color").default_value = (0.8, 0.8, 0.8, 1.0)    # Corresponds to Cycles diffuse
+        self.inputs.new('NodeSocketFloatFactor', "Roughness").default_value = 1.0
```

Comment:
Lack of soft/hard limits, editing this input in UI with mouse sets value to unreasonable numbers, for example "-3997".

Reply:
BTW rpr diffuse uses this.